src/main/java/org/broadinstitute/hellbender/utils/io/TextFeatureReader.java

```diff
@@ -0,0 +1,244 @@
+            } catch ( final IOException ioe ) {
+                throw new UserException("Error while checking for existence of index for " + path);
+            }
+        }
+        return queryable;
+    }
+
+    @Override
+    public FeatureIterator<F> query( final String chr, final int start, final int end ) {
+        if ( index == null ) {
+            if ( !isQueryable() ) {
+                throw new UserException("Querying non-queryable feature reader for " + path);
+            }
+            index = IndexUtils.loadTabixIndex(path);
+            if ( index == null ) {
+                throw new UserException("Despite its appearing to exist, couldn't load index for " + path);
+            }
+        }
+        final List<Block> blocks = index.getBlocks(chr, start, end);
+        return new FeatureIterator<>(path,
+                codec,
+                new SimpleInterval(chr, start, end),
+                blocks == null || blocks.isEmpty() ? BOGUS_FILE_OFFSET : blocks.get(0).getStartPosition());
+    }
+
+    @Override
+    public void close() {}
+
+    @Override
+    public List<String> getSequenceNames() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public Object getHeader() {
+        if ( header == null ) {
+            final LineIterator lineIterator = new LineIterator(path);
+            try {
+                header = codec.readHeader(lineIterator);
+            } catch ( final IOException ioe ) {
+                throw new UserException("Can't read header from " + path, ioe);
+            }
+            lineIterator.close();
+        }
+        return header.getHeaderValue();
+    }
+
+    private final static class LineIterator implements Iterator<String> {
+        private final Path path;
+        private final BufferedReader reader;
+        private String nextLine;
+
+        public LineIterator( final Path path ) {
+            this.path = path;
+            try {
+                InputStream in = new BufferedInputStream(Files.newInputStream(path), 256*1024);
+                if ( path.toString().endsWith(".gz") ) {
+                    in = new BlockCompressedInputStream(in);
+                }
+                reader = new BufferedReader(new InputStreamReader(in));
+            } catch ( final IOException ioe ) {
+                throw new UserException("Can't open " + path, ioe);
+            }
+            advance();
+        }
+
+        public LineIterator( final Path path, final long virtualFileOffset ) {
+            this.path = path;
+            try {
+                final SeekableByteChannel sbc = Files.newByteChannel(path);
+                sbc.position(virtualFileOffset >> 16);
+                final InputStream is =
+                        new BlockCompressedInputStream(
+                                new BufferedInputStream(Channels.newInputStream(sbc), 64*1024));
+                is.skip(virtualFileOffset & 0xffffL);
```

Comment:
I think this is allowed to skip less than you specified.  You should probably check the return value and throw if it's off.